rali/rali/source/meta_node_rotate.cpp

```diff
@@ -0,0 +1,133 @@
+void RotateMetaNode::initialize()
+{
+    _src_height_val.resize(_batch_size);
+    _src_width_val.resize(_batch_size);
+    _angle_val.resize(_batch_size);
+}
+float calculate_rotate_point_x(float angle, float x, float cx, float cy)
+{
+    float rotate[2];
+    rotate[0] = cos(RAD(angle));
+    rotate[1] = -1 * sin(RAD(angle));
+    float trans_x = cx + (rotate[0] * cx) + (rotate[1] * cy);
+    return trans_x;
+}
+float calculate_rotate_point_y(float angle, float y, float cx, float cy)
+{
+    float rotate[2];
+    rotate[0] = sin(RAD(angle));
+    rotate[1] = cos(RAD(angle));
+    float trans_y = cy + (rotate[0] * cx) + (rotate[1] * cy);
+    return trans_y;
+}
+void RotateMetaNode::update_parameters(MetaDataBatch* input_meta_data)
+{
+    initialize();
+    if(_batch_size != input_meta_data->size())
+    {
+        _batch_size = input_meta_data->size();
+    }
+    _src_width = _node->get_src_width();
+    _src_height = _node->get_src_height();
+    _dst_width = _node->get_dst_width();
+    _dst_height = _node->get_dst_height();
+    _angle = _node->get_angle();
+    vxCopyArrayRange((vx_array)_src_width, 0, _batch_size, sizeof(uint),_src_width_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_src_height, 0, _batch_size, sizeof(uint),_src_height_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    vxCopyArrayRange((vx_array)_angle, 0, _batch_size, sizeof(float),_angle_val.data(), VX_READ_ONLY, VX_MEMORY_TYPE_HOST);
+    for(int i = 0; i < _batch_size; i++)
+    {
+        auto bb_count = input_meta_data->get_bb_labels_batch()[i].size();
+        int labels_buf[bb_count];
+        float coords_buf[bb_count*4];
+        memcpy(labels_buf, input_meta_data->get_bb_labels_batch()[i].data(),  sizeof(int)*bb_count);
+        memcpy(coords_buf, input_meta_data->get_bb_cords_batch()[i].data(), input_meta_data->get_bb_cords_batch()[i].size() * sizeof(BoundingBoxCord));
+        BoundingBoxCords bb_coords;
+        BoundingBoxCord temp_box;
+        temp_box.x = temp_box.y = temp_box.w = temp_box.h = 0;
+        BoundingBoxLabels bb_labels;
+        BoundingBoxCord dest_image;
+        dest_image.x = 0;
+        dest_image.y = 0;
+        dest_image.w = _dst_width;
+        dest_image.h = _dst_height;
+        for(uint j = 0, m = 0; j < bb_count; j++)
+        {
+            BoundingBoxCord box;
+            float src_bb_x, src_bb_y, bb_w, bb_h, dest_bb_x, dest_bb_y;
+            float cx, cy;
+            float x1, y1, x2, y2, x3, y3, x4, y4, min_x, min_y, max_x, max_y;
+            cx = _dst_width / 2;
+            cy = _dst_height / 2;
+            src_bb_x = (coords_buf[m++]);
+            src_bb_y = (coords_buf[m++]);
+            bb_w = (coords_buf[m++]);
+            bb_h = (coords_buf[m++]);
+            dest_bb_x = (cx) - ((_src_width_val[i] / 2) - src_bb_x);
+            dest_bb_y = (cy) - ((_src_height_val[i] / 2) - src_bb_y);
+            x1 = calculate_rotate_point_x(dest_bb_x, _angle_val[i], cx, cy);
```

Comment:
try to compute all coordinates which use the same angle value in the same function for better performance